### PR TITLE
Add detailed error message to `IrreversibleMigration`

### PR DESCRIPTION
**Pull request by yui-knk**

### CODE REVIEW

File: activerecord/lib/active_record/migration/command_recorder.rb [hunk 1]
```diff
@@ -69,7 +69,7 @@ def record(*command, &block)
       # invert the +command+.
       def inverse_of(command, args, &block)
         method = :"invert_#{command}"
-        raise IrreversibleMigration unless respond_to?(method, true)
+        raise IrreversibleMigration, "#{method} is not defined, so #{command} is not reversible" unless respond_to?(method, true)
```
Comment (rafaelfranca) [inline activerecord/lib/active_record/migration/command_recorder.rb]: I don't think we should talk about the `method` here. It is something internal to Rails, users don't need to care about it. Maybe something like: "This migration uses #{command} that is not automatically reversible, please ensure you have a `down` method"?

Comment (yui-knk) [inline activerecord/lib/active_record/migration/command_recorder.rb]: Oh thank you I will rewrite later !

**Comment by yui-knk** [comment]
Updated and I add `Gemfile.lock` mistakenly. I will remove `Gemfile.lock` when I squash these commits.

File: activerecord/lib/active_record/migration/command_recorder.rb [hunk 2]
```diff
@@ -69,7 +69,7 @@ def record(*command, &block)
       # invert the +command+.
       def inverse_of(command, args, &block)
         method = :"invert_#{command}"
-        raise IrreversibleMigration unless respond_to?(method, true)
+        raise IrreversibleMigration, "This migration uses #{command} that is not automatically reversible, please define down method" unless respond_to?(method, true)
```
Comment (senny) [inline activerecord/lib/active_record/migration/command_recorder.rb]: If we suggest defining a `down` method, this means that the user also needs to change `def change` to `def up`.

Another option would be a `reversible do |dir|` block.

**Comment by senny** [comment]
It's a good idea to make the exception message more descriptive. At the same time we should improve the rDocs of [`IrreversibleMigration`](http://edgeapi.rubyonrails.org/classes/ActiveRecord/IrreversibleMigration.html). Let's put some examples there how you can mitigate the problem.

1. by using `up` and `down`
2. by using `reversible do |dir|`

File: activerecord/lib/active_record/migration/command_recorder.rb (hunk 2 above)
Comment (yui-knk) [inline activerecord/lib/active_record/migration/command_recorder.rb]: I forgot `reversible do |dir|` :cry: 
I will rewrite once more :)

**Comment by yui-knk** [comment]
@senny  updated!

File: Gemfile.lock [hunk 3]
```diff
@@ -202,6 +202,10 @@ GEM
     mysql2 (0.3.19)
     nokogiri (1.6.6.2)
       mini_portile (~> 0.6.0)
+    nokogiri (1.6.6.2-x64-mingw32)
+      mini_portile (~> 0.6.0)
+    nokogiri (1.6.6.2-x86-mingw32)
+      mini_portile (~> 0.6.0)
```
Comment (senny) [inline Gemfile.lock]: I don't think this belongs into the patch.

Comment (yui-knk) [inline Gemfile.lock]: I commented on conversation,

> Updated and I add Gemfile.lock mistakenly. I will remove Gemfile.lock when I squash these commits.

:)

**Comment by senny** [comment]
@yui-knk thanks :yellow_heart: I made some minor modifications in the merge commit.

Sadly I forgot to check wether the commits are squashed :sweat:, please amend review changes directly to keep the PR in a mergable state at all times. This helps reviewers a lot.

**Comment by yui-knk** [comment]
@senny Thanks for your review and merge. And sorry to my mistake :cry: 
I confused rails way with [this](https://github.com/thoughtbot/guides/tree/master/code-review#having-your-code-reviewed).

> Do not squash until the branch is ready to merge. Reviewers should be able to read individual updates based on their earlier feedback.

I will be careful from now on :)

**Comment by senny** [comment]
@yui-knk could you point me to the place where you found that sentence?

**Comment by yui-knk** [comment]
6th of list item of "Having Your Code Reviewed".
But this is thoughtbot/guides (not rails guide).

I found nothing related with "Do not squash" in Rails Guide.

**Comment by jbilbo** [comment]
@senny @yui-knk It makes sense for big PRs but I don't think it's practical with small ones like this. 
But yeah... @rails-bot also recommends it, for example: https://github.com/rails/rails/pull/21502#issuecomment-137849198
> If any changes to this PR are deemed necessary, please add them as extra commits. This ensures that the reviewer can see what has changed since they last reviewed the code

**Pull request merged**